test/compression.js

```diff
@@ -301,6 +301,95 @@ describe('compression()', function () {
       .expect(200, done)
   })
 
+  it('should support removeListener("drain"); stream present', function (done) {
+    // compression doesn't proxy listenerCount() to the compression stream, so
+    // instead watch for a MaxListenersExceededWarning
+    var hasWarned = false
+    var onWarning = function (warning) {
+      hasWarned = true
+    }
+    process.on('warning', onWarning)
+    var server = createServer({threshold: 0}, function (req, res) {
+      res.setHeader('Content-Type', 'text/plain')
+      var len = bytes('40kb')
+      var buf = Buffer.alloc(len, '.')
+      res.write(buf)
+      // (12 is greater than the default number of max listeners)
```

Comment:
if we're relying on the warning, probably should set the max listener count through the api so we know it will trigger even if node.js changes the default value.